scripts/ci/check_compliance.py

```diff
@@ -220,6 +220,50 @@ def run(self):
                 self.failure(output)
 
 
+class DevicetreeBindingsCheck(ComplianceTest):
```

Comment:
I'm not going to NAK over this because this is already a very altruistic pull request, and this kinda works, but it is duplicating features that we already have elsewhere in a slightly worse way.

See e.g.

```py
    bindings = edtlib.bindings_from_paths(binding_files, ignore_errors=True)
```

in gen_devicetree_rest.py, which relies on the edtlib abstraction boundary to find the bindings, and supports YAML files with both .yml and .yaml extensions.

If you used this, you'd get a bunch of first-class `Binding` objects, and you could use `binding.raw` values to look and see if anything in `raw['properties']` has an explicit `required: true` instead of just checking if the string `"required: false"` appears in the text of the file. The latter string-based check is subject to false positives if the string appears in a YAML comment, for example.

Something to improve in the future, though, if we just want to get this in now.